lib/saml/user.rb

```diff
@@ -17,35 +17,37 @@ def last_signed_in
       Time.current.utc
     end
 
-    def loa
-      { current: loa_current, highest: loa_highest(attributes) }
-    end
-
     def to_hash
       Hash[serializable_attributes.map { |k| [k, @decorated.send(k)] }]
     end
 
+    def authn_context
+      return 'dslogon' if dslogon?
+      return 'mhv' if mhv?
+      nil
+    end
```

Comment:
Seems like you're overloading the term `authn_context`.  Perhaps rename this method to `def identity_source`?  Also, it looks to return an enum type which should probably be defined somewhere.

Reply:
So not really, this value gets passed to User.from_saml and that's how user gets initialized with the authn_context. See also that it is in the serializable attributes?

But to Patrick's point, it might also be in the REXML document and I could possibly pull it out from there the same manner we're doing on idme decorator.